geps/gep-2627/index.md

```diff
@@ -0,0 +1,50 @@
+# GEP-2627: DNS configuration within Gateway API
+
+* Issue: [#2627](https://github.com/kubernetes-sigs/gateway-api/issues/2627)
+* Status: Provisional
+
+## TLDR
+
+For gateway infrastructure to be valuable we need to be able to connect clients to these gateways. A common way to achieve this is to use domain names/hostnames and DNS. Gateways define listeners that can have assigned hostnames or wildcards.  The guidelines for DNS configuration are a critical piece of service networking, but this is currently not expressible as part of Gateway API.   Instead of leaving this as an exercise for the user to figure out, this proposal attempts to provide options to ease Gateway API operations.
+
+## Goals
+* Provide a way for Gateway API resource owners to mark their resources as relevant for external DNS provisioning
+* Ensure that the above method has a way for multiple providers to be present in the cluster and be able to actuate external DNS provisioning requests
+* Ensure that any method is based on structured fields and makes the most of `status` on whatever resources are relevant, whether they are existing Gateway API resources or new resources.
+* Increase portability and supportability between Gateway API implementations and third party controllers offering DNS integration.
+* Clarity on the scope of hostnames under management. (This should _not_ be able to be used to affect the standard in-cluster DNS configuration)
+
+## Non-Goals
+
+* Anything to do with configuring in-cluster DNS. This support is for configuration outside the cluster only.
```

Comment:
I'm not opposed to this as a non-goal, except I was wondering if you would please elaborate a bit more on what's driving you to leave it out?

Reply:
As mentioned above. This could prob be clarified more as it isn't that we can't handle host names and IPs that are not for external use but rather that we wont interfer with things like the service discovery and existing DNS infrastructure in k8s

Reply:
I was considering this overnight and I'm actually starting to think that we should in fact remove this non-goal. My thinking is that since we're pretty much just a spec and status where implementations can use whatever integration (cloud or otherwise) they want to fulfill it and report the status, then in-cluster really shouldn't matter too much from our perspective (except that we _might_ want to provide spec delineation so that a user can request "this DNS name, ONLY intra-cluster", or something like that).

```suggestion
```

Can you be persuaded? Or did you have some strong feelings that it would be a detriment to include in-cluster DNS now?

Reply:
Yes. I can be persuaded. Coming back to this, I think you are correct and this may have been added based on some questions around security. As you say, this is proposing a spec and defined status and as such can be implemented by multiple providers. So ultimately it is down to the cluster admin and the RBAC system to decide what implementation to use and who they want to allow to create this API. 
One thought though is around the target ref, I guess I had initially only thought of it as targeting a gateway initially are you thinking beyond that ? such as a service ?

Reply:
Well, I think we need to consider the user stories (see https://github.com/kubernetes-sigs/gateway-api/pull/2712#discussion_r1951304911) and decide who needs this functionality, and if its safe to provide them.

I think if you're OK with committing the suggestion above, we can consider this comment resolved and then can follow up in that other comment.